### ISSUE TITLE
Habitat integration issues

### ISSUE THREAD

**Issue by dcorrea78** [bsileo/hubitat_poolcontroller#22]
I posted this over on the habitat forums a few days ago but I'm looking for help getting the nodejs pool controller integration working properly with my Pentair Easytouch 8 system. I successfully installed the nodejs pool controller integration through the package manager and it is connected to my Screenlogic Easy touch 8 via the nodejs pool controller running on my Rpi4.

The issue I am running into right now is that I am trying to make a dashboard to control my pool equipment but I cannot figure out:

How do I create a "RPM" control switch
How do i get the dashboard to poll the pool equipment and update accordingly.
For the RPM control, I created two virtual buttons; one for RPM up and one for RPM down. The problem is that I cannot find any attribute/button/switch etc in the Pentair intelliflo device, pool control device or easytouch device that increases RPM. Any ideas on where to find that and what type of device template I need to set the hubitat tile to in order to get it to work?

Secondly, the hubitat integration doesn't seem to poll the pool equipment and the hubitat devices get out of sync with the actual equipment.
For example, I have a tile that reads the pump RPM using the "attribute" function of the tile wizard. However, if I change the pump RPM via the poolcontroller dashboard, the tile and hubitat device do not update accordingly. I have to manually refresh the device to get the rpm updated.

How do I ensure that any action taken outside of the hubitat environment (nodejs poolcontrol dashboard, Screenlogic app, etc) properly update the hubitat states?

**Comment by philipfoltz** [comment]
I'm struggling with the same issue of not getting updates from the poolcontroller to hubitat. I have the intigration set in the poolcontroller but not sure if I have the settings correct. Hubitat will send updates to the poolcontroller. I'm at a loss.

**Comment by bsileo** [comment]
Did you setup the Event Interface in the configuration file on the
PoolController to point it at your Hubitat hub?

https://github.com/bsileo/hubitat_poolcontroller#setup-the-event-interface

**Comment by philipfoltz** [comment]
Yes, but there is also a protocol in that setting. Not sure which to choose.On May 19, 2023, at 8:49 PM, Brad Sileo ***@***.***> wrote:﻿
Did you setup the Event Interface in the configuration file on the
PoolController to point it at your Hubitat hub?

https://github.com/bsileo/hubitat_poolcontroller#setup-the-event-interface

**Comment by bsileo** [comment]
I'm not sure about the protocol bit - i am a little behind on NJsPC
versions - maybe that is something new in V8?

Are you seeing anything in the Hubitat logs to indicate that traffic is
coming from the event interface to the Hubitat hub?

**Comment by dcorrea78** [comment]
The protocol options are http, https, or mqtt. I am using http since I don't believe Hubitat supports https.

**Comment by dcorrea78** [comment]
I just realized that I didn't re-enable the integration after re-installing the pool-controller from scratch. I will check to see if Hubitat is getting the device status properly later today. 

As for controlling the RPM, do you know where I can find the attribute that controls rpm speed? Basically I would like to mimic the poolcontroller-dashPanel controller on the hubitat dashboard and use a button to increase or decrease rpm.

**Comment by bsileo** [comment]
> The protocol options are http, https, or mqtt. I am using http since I don't believe Hubitat supports https.

Ah yes HTTP would be the correct option here.

**Comment by dcorrea78** [comment]
I just checked, and I still have to manually refresh the devices in hubitat to get the status updated. The hubitat integration also gives an option for the type of connection. It is defaulted to rest but http is an option there as well. I just changed to http to see what will happen, but can you confirm what it should be?

**Comment by bsileo** [comment]
> I just checked, and I still have to manually refresh the devices in hubitat to get the status updated. The hubitat integration also gives an option for the type of connection. It is defaulted to rest but http is an option there as well. I just changed to http to see what will happen, but can you confirm what it should be?

Yes sorry - I think that should be rest.  I would have to catch up on versions to verify this in my install.

**Comment by bsileo** [comment]
> I just realized that I didn't re-enable the integration after re-installing the pool-controller from scratch. I will check to see if Hubitat is getting the device status properly later today.
> 
> As for controlling the RPM, do you know where I can find the attribute that controls rpm speed? Basically I would like to mimic the poolcontroller-dashPanel controller on the hubitat dashboard and use a button to increase or decrease rpm.

There is not a command on the Hubitat side to set the pump RPM at this time.  This could be added to the [pump device](https://github.com/bsileo/hubitat_poolcontroller/tree/master/devicetypes/bsileo/pool-controller-pump.src) I think, though I am not sure if that API exists on NJSPC at present.

**Comment by bsileo** [comment]
> > I just checked, and I still have to manually refresh the devices in hubitat to get the status updated. The hubitat integration also gives an option for the type of connection. It is defaulted to rest but http is an option there as well. I just changed to http to see what will happen, but can you confirm what it should be?
> 
> Yes sorry - I think that should be rest. I would have to catch up on versions to verify this in my install.

What is your port set to for Hubitat? This should be 39501

**Comment by philipfoltz** [comment]
Here are my settings in the Pool Controller. 

 "hubitat": {
        "name": "Hubitat",
        "type": "http",
        "enabled": true,
        "fileName": "smartThings-Hubitat.json",
        "globals": {},
        "options": {
          "host": "192.168.150",
          "port": 39501,
          "protocol": "http://"
        }

![Pool Controller Settings 1](https://github.com/bsileo/hubitat_poolcontroller/assets/51804223/1d17c65d-1492-4273-ba5c-0c4fb724c9c4)

**Comment by bsileo** [comment]
That's not a valid IP address for your Hubitat.

**Comment by philipfoltz** [comment]
Well now I feel stupid.. ;) I have updated it and testing

**Comment by philipfoltz** [comment]
Still not getting updates from the Pool Controller to Hubiat but Hubitat will control the Pool Controller.

**Comment by philipfoltz** [comment]
Where is this file - "smartThings-Hubitat.json"? When I search for it nothing is found.

**Comment by bsileo** [comment]
https://github.com/rstrouse/nodejs-poolController/blob/master/web/bindings/smartThings-Hubitat.json

**Comment by philipfoltz** [comment]
I found it. Just wanted to make sure it was there.  And it matches

**Comment by bsileo** [comment]
Set the IDE Logging Level on your Pool Controller device in Hubitat to
"Debug" and see what you get in logging.

![image](https://github.com/bsileo/hubitat_poolcontroller/assets/6116016/6b8e6a89-9d9c-4a8b-bf5d-fce301c733d1)

**Comment by bsileo** [comment]
Here you can see mine with all the event data flowing into Hubitat

![image](https://github.com/bsileo/hubitat_poolcontroller/assets/6116016/f31cc36c-2c39-4164-adaf-31d6e85d2b44)

**Comment by philipfoltz** [comment]
Here is what I'm getting

![Logs](https://github.com/bsileo/hubitat_poolcontroller/assets/51804223/cab32672-333c-4da5-9384-229c2ce1bbe1)

**Comment by bsileo** [comment]
Certainly, no inbound events showing up there. You have to take a look at the logs on the NJSPC side to see if you can find these events being sent to Hubitat. 

Set the Log Level:
![image](https://github.com/bsileo/hubitat_poolcontroller/assets/6116016/7d5c8a83-1cb7-4777-bd5c-483237afd593)

Watch the Logs directory for output under consoleLog\*.log:
[5/20/2023, 10:55:14 AM] debug: Sending [temps] request to Hubitat: {"headers":{"CONTENT-TYPE":"application/json","X-EVENT-TYPE":"temps"},"host":"192.168.2.106","port":39501,"method":"NOTIFY","path":"/notify"}
[5/20/2023, 10:55:22 AM] debug: Sending [chlorinator] request to Hubitat: {"headers":{"CONTENT-TYPE":"application/json","X-EVENT-TYPE":"chlorinator"},"host":"192.168.2.106","port":39501,"method":"NOTIFY","path":"/notify"}
[5/20/2023, 10:55:22 AM] debug: Sending [chlorinator] request to Hubitat: {"headers":{"CONTENT-TYPE":"application/json","X-EVENT-TYPE":"chlorinator"},"host":"192.168.2.106","port":39501,"method":"NOTIFY","path":"/notify"}
[5/20/2023, 10:55:24 AM] debug: Sending [temps] request to Hubitat: {"headers":{"CONTENT-TYPE":"application/json","X-EVENT-TYPE":"temps"},"host":"192.168.2.106","port":39501,"method":"NOTIFY","path":"/notify"}
[5/20/2023, 10:55:34 AM] debug: Packet not processed: 255,0,255,165,1,16,34,215,1,1,1,177
[5/20/2023, 10:55:34 AM] debug: Packet not processed: 255,255,255,255,255,0,255,165,1,15,16,23,16,1,0,0,0,0,0,0,0,0,0,128,128,0,0,0,128,2,109

**Comment by philipfoltz** [comment]
Where do I find this log?

**Comment by bsileo** [comment]
Under the directory where you installed NJSPC - something like ~/nodejs-poolController/logs

**Comment by philipfoltz** [comment]
Got it.

**Comment by philipfoltz** [comment]
I see it sending, still deciphering

**Comment by dcorrea78** [comment]
When I get home I'll check my logs as well

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by philipfoltz** [comment]
Ok, after a few restarts and reading the logs it is now working. Thank you for your help! Awesome work on your integrations!

**Comment by bsileo** [comment]
Ok great news @philipfoltz 

What about you @dcorrea78 ?  I will close this for now but can reopen if you are still having issues. 

I actually had not looked at this in about a year - always cool when I hear from others getting use out of it!

**Comment by dcorrea78** [comment]
I verified the  eventmanager and the web bindings code were correct. 

> "hubitat": {
        "name": "Hubitat",
        "type": "rest",
        "enabled": true,
        "fileName": "smartThings-Hubitat.json",
        "globals": {},
        "options": {
          "host": "192.168.1.132",
          "port": 39501,
          "protocol": 

> "http://"

<{

  "context": {
    "name":  "SmartThings",
    "vars": {},
    "options": {
      "method": "NOTIFY",
      "path": "/notify",
      "headers": {
        "CONTENT-TYPE": "application/json",
        "X-EVENT-TYPE": "@bind=eventName;"
      }
    }
  },
  "events": [
    {
      "name": "justAnExample",
      "description":  "Shows how the binding can be used to send variable data to SmartThings.  This can be any combination of data.",
      "body": "{\"air\":\"@bind=data.air;@bind=data.units.name.substring(0, 1);\", \"total\":@bind=data.solar + data.waterSensor1;, \"status\":\"@bind=state.equipment.model;\"}",
      "options": {
        "headers": {
          "X-EVENT2": ""
        }
      }
    },
    {
        "name": "config",
        "description": "Not used for updates",
        "enabled": false
    },
    { "name": "*", "description": "All events that are not trapped by other event names.  Sends the entire emitted response.", "body": "@bind=data;" }
  ]>

I turned on debug for the easytouch and this is what I see:
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:24:12.723 PM[error](https://192.168.1.84/logs?tab=past&deviceId=1321#)Ciurcuit update failed with code 408
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:24:02.669 PM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Send PUT to /state/circuit/setState with [id:6, state:0] and setCircuitCallback
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:24:02.667 PM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Executing setCircuit with Pool - 6 to 0
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:24:02.652 PM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Got OFF from Pool
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.870 PM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Unhandled Temperature Element 'equipmentType' : temps
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.784 PM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Parse Temps 200 -- true
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.755 PM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Send GET to with [uri:http://192.168.1.132:4200, path:/config/all, requestContentType:application/json, contentType:application/json, body:] CB=parseConfigAll
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.751 PM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Send GET to with [uri:http://192.168.1.132:4200, path:/state/temps, requestContentType:application/json, contentType:application/json, body:] CB=parseTemps
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.749 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Light Group 192 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.738 PM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Create/Update Light Children for this device
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.737 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)No Intellichem devices found on Controller
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.735 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Pool Chlorinator Chlorinator 1 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.724 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Feature Switch for Feature 8 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.717 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Feature Switch for Feature 7 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.710 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Feature Switch for Feature 6 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.703 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Feature Switch for Feature 5 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.696 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Feature Switch for Feature 4 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.689 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Feature Switch for High Speed and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.682 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Feature Switch for Water Bowls and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.673 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Feature Switch for Bubbler and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.663 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Circuit for AUX 7 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.657 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Circuit for AUX 6 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.650 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Circuit for AUX 5 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.643 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Circuit for AUX 4 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.636 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Circuit for AUX 3 and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.629 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Circuit for Air Blower and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.622 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Circuit for Lights and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.615 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Circuit for Spa and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.608 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Circuit for Pool and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.598 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Heater called Solar Heater and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.591 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Heater called Gas Heater and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.578 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Pump called Intelliflo VS and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.566 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Body called Spa and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.559 PM[info](https://192.168.1.84/logs?tab=past&deviceId=1321#)Found existing Body called Pool and updated it
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.526 PM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Pool Controller manageChildren starting
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 12:18:55.480 PM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Send GET to with [uri:http://192.168.1.132:4200, path:/config, requestContentType:application/json, contentType:application/json, body:] CB=configurationCallback
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 09:15:03.230 AM[error](https://192.168.1.84/logs?tab=past&deviceId=1321#)Ciurcuit update failed with code 408
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 09:14:53.193 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Send PUT to /state/circuit/setState with [id:6, state:1] and setCircuitCallback
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 09:14:53.192 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Executing setCircuit with Pool - 6 to 1
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 09:14:53.176 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Got ON Request from Pool
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:31:17.024 AM[error](https://192.168.1.84/logs?tab=past&deviceId=1321#)Ciurcuit update failed with code 408
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:31:06.983 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Send PUT to /state/circuit/setState with [id:6, state:0] and setCircuitCallback
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:31:06.981 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Executing setCircuit with Pool - 6 to 0
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:31:06.980 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Got OFF from Pool
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:31:04.132 AM[error](https://192.168.1.84/logs?tab=past&deviceId=1321#)Ciurcuit update failed with code 408
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:30:54.096 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Send PUT to /state/circuit/setState with [id:6, state:1] and setCircuitCallback
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:30:54.094 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Executing setCircuit with Pool - 6 to 1
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:30:54.092 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Got ON Request from Pool
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:28:30.798 AM[error](https://192.168.1.84/logs?tab=past&deviceId=1321#)Ciurcuit update failed with code 408
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:28:20.763 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Send PUT to /state/circuit/setState with [id:6, state:0] and setCircuitCallback
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:28:20.761 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Executing setCircuit with Pool - 6 to 0
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:28:20.759 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Got OFF from Pool
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:28:14.914 AM[error](https://192.168.1.84/logs?tab=past&deviceId=1321#)Ciurcuit update failed with code 408
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:28:04.879 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Send PUT to /state/circuit/setState with [id:6, state:1] and setCircuitCallback
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:28:04.877 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Executing setCircuit with Pool - 6 to 1
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:28:04.875 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Got ON Request from Pool
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:27:52.754 AM[error](https://192.168.1.84/logs?tab=past&deviceId=1321#)Ciurcuit update failed with code 408
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:27:42.714 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Send PUT to /state/circuit/setState with [id:6, state:0] and setCircuitCallback
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:27:42.713 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Executing setCircuit with Pool - 6 to 0
[dev:1321](https://192.168.1.84/logs?tab=past&deviceId=1321#)2023-05-20 08:27:42.696 AM[debug](https://192.168.1.84/logs?tab=past&deviceId=1321#)Got OFF from Pool>

I'm not sure what else to check. In the debug logs, the power off command came from me hitting the power tile on the hubitat dashboard to make it match the actual poolcontroller.

**Comment by dcorrea78** [comment]
Ok looks like I got it working also. I didn't have the poolcontroller looking at the right IP for my hubitat. Doh!

I am seeing an error in the logs for the heater though in case you want to check that out:
![image](https://github.com/bsileo/hubitat_poolcontroller/assets/34433189/6c59ff03-24f1-4651-a003-634548d8121f)

**Comment by philipfoltz** [comment]
I’m also interested in heater control, is that an option?On May 20, 2023, at 11:53 AM, dcorrea78 ***@***.***> wrote:﻿
Ok looks like I got it working also. I didn't have the poolcontroller looking at the right IP for my hubitat. Doh!
I am seeing an error in the logs for the heater though in case you want to check that out:

**Comment by bsileo** [comment]
Heater control is available. The Body device exposes heater / setpoint commands. It has the ThermostateHeatingSetpoint capability which lets you control it with Dashboards, etc.

**Comment by philipfoltz** [comment]
Great! I’ll review that!On May 21, 2023, at 8:07 PM, Brad Sileo ***@***.***> wrote:﻿
Heater control is available. The Body device exposes heater / setpoint commands. It has the ThermostateHeatingSetpoint capability which lets you control it with Dashboards, etc.

**Comment by philipfoltz** [comment]
> Heater control is available. The Body device exposes heater / setpoint commands. It has the ThermostateHeatingSetpoint capability which lets you control it with Dashboards, etc.

I've been poking around this but can't quite figure it out. I see where the points are exposed in the device but can't get a dashboard tile to control them...

**Comment by bsileo** [comment]
> > Heater control is available. The Body device exposes heater / setpoint commands. It has the ThermostateHeatingSetpoint capability which lets you control it with Dashboards, etc.
> 
> I've been poking around this but can't quite figure it out. I see where the points are exposed in the device but can't get a dashboard tile to control them...

What Dashboard are you using?  Here is an example of this tile deployed in Sharptools:

![image](https://github.com/bsileo/hubitat_poolcontroller/assets/6116016/bb58874b-0b3a-477c-acf3-db07c8bc871d)

**Comment by philipfoltz** [comment]
As of right now just the Hubitat dashboards. Probably need to make the jump to a more robust dashboard option.

**Comment by bsileo** [comment]
Ok I don't have those installed so I am not sure how they support various Thermostat configurations.  We are using the standard capability per https://docs2.hubitat.com/developer/driver/capability-list - capability.thermostatHeatingSetpoint

**Comment by philipfoltz** [comment]
Thank you for this feedback!Sent from my iPhonePhilip Foltz On Jun 7, 2023, at 8:29 AM, Brad Sileo ***@***.***> wrote:﻿
Ok I don't have those installed so I am not sure how they support various Thermostat configurations.  We are using the standard capability per https://docs2.hubitat.com/developer/driver/capability-list - capability.thermostatHeatingSetpoint

**Comment by philipfoltz** [comment]
I'm playing with sharptools now. How do you turn the heater on and off, I have the setpoint working. That attribute does not seem to be listed in sharptools.

**Comment by bsileo** [comment]
Check out the CurrentHeaterMode attribute on the Pool Body to see its state. I am using a [Custom Rule](https://sharptools.io/rule/manage) in SharpTools to change the HeaterMode.  This calls the command nextHeaterMode.  The challenge here is that there can be different heater modes available for different installations (e.g. some have Solar Pref, etc)

![image](https://github.com/bsileo/hubitat_poolcontroller/assets/6116016/631cc3d6-309a-451c-ab07-a08eda0f3a53)

**Comment by philipfoltz** [comment]
That will help me greatly! Thank you for this insight!

**Comment by dcorrea78** [comment]
Philip,

I have the heater control working in the standard Hubitat dashboard with a
combination of virtual buttons and rules.

I created virtual buttons for pool heater temp up, pool heater temp down,
spa heater temp up and spa heater temp down.
I then created a rule for each action. For example, for pool heater temp up
my rule is as follows:

*Trigger event *- When pool heater set temp up button 1 is pushed
*Actions - * Capability of action device = switch
                Select switches to act on = Pool (the pool device on my
system contains the setPointUp attribute for the heater)
               Select custom command = setPointUp

I repeated this for the other Heater and Spa heater actions.

On the dashboard I added a tile for each "button" with the device set to
the virtual button I want to trigger and the template set as a button. I
also set the button number to 1 and the button command to push. I also
added a tile that polls the heatingSetpoint attribute of the heater so that
the set temp is displayed.

[image: image.png]

**Comment by philipfoltz** [comment]
Thank you! I’ve been tinkering with these and your feedback is very helpful.Sent from my iPhonePhilip Foltz On Jun 7, 2023, at 12:17 PM, dcorrea78 ***@***.***> wrote:﻿
Philip,

I have the heater control working in the standard Hubitat dashboard with a
combination of virtual buttons and rules.

I created virtual buttons for pool heater temp up, pool heater temp down,
spa heater temp up and spa heater temp down.
I then created a rule for each action. For example, for pool heater temp up
my rule is as follows:

*Trigger event *- When pool heater set temp up button 1 is pushed
*Actions - * Capability of action device = switch
                Select switches to act on = Pool (the pool device on my
system contains the setPointUp attribute for the heater)
               Select custom command = setPointUp

I repeated this for the other Heater and Spa heater actions.

On the dashboard I added a tile for each "button" with the device set to
the virtual button I want to trigger and the template set as a button. I
also set the button number to 1 and the button command to push. I also
added a tile that polls the heatingSetpoint attribute of the heater so that
the set temp is displayed.

[image: image.png]
